lib/bandit.ex

```diff
@@ -303,6 +303,8 @@ defmodule Bandit do
     |> Keyword.get(:plug)
     |> case do
       nil -> raise "A value is required for :plug"
+      {plug_fn, plug_options} when is_function(plug_fn) -> {plug_fn, plug_options}
+      plug_fn when is_function(plug_fn) -> {plug_fn, []}
       {plug, plug_options} -> {plug, plug.init(plug_options)}
       plug -> {plug, plug.init([])}
```

Comment:
Should we us `is_function(plug_fn, 2)` to ensure the function has the correct arity?
Maybe checking if the plug is an atom here would be good as well (or even if the Module exists / is loaded), like:

```elixir
case do
  nil ->
    raise "A value is required for :plug"

  {plug_fn, plug_options} when is_function(plug_fn, 2) ->
    {plug_fn, plug_options}

  plug_fn when is_function(plug_fn, 2) ->
    {plug_fn, []}

  {plug, plug_options} when is_atom(plug) ->
    unless function_exported?(plug, :call, 2) do
      raise "Invalid plug module: #{inspect(plug)}"
    end

    {plug, plug.init(plug_options)}

  plug when is_atom(plug) ->
    unless function_exported?(plug, :call, 2) do
      raise "Invalid plug module: #{inspect(plug)}"
    end

    {plug, plug.init([])}

  _other ->
    raise "Invalid value for :plug"
end
```

This would error on starting the server vs. error on every request, if the function or plug are (obviously) invalid.

Reply:
Yes we should! Lemme get this in.

Reply:
Updated!